src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -418,7 +418,8 @@ public override string ToString()
             return DiagnosticFormatter.Instance.Format(this, CultureInfo.CurrentUICulture);
         }
 
-        public abstract override bool Equals(object? obj);
+        public sealed override bool Equals(object? obj)
```

Comment:
>sealed [](http://example.com/codeflow?start=15&length=6)

Can `Diagnostic` be implemented outside of `Microsoft.CodeAnalysis`? If so, does this change need to be documented?

Reply:
I think this is a breaking change for people who subclassed `Diagnostic`, so it should be documented.

Separately, I wonder who subclasses Diagnostic but doesn't have an IVT to the compiler? I imagine EE has an IVT to the compiler. Should/can we introduce an internal constructor on this type to prevent others from subclassing it in the future?

Reply:
This cannot be implemented by anyone but compiler Team.  Tyep already contains internal abstracts:

```
internal abstract Diagnostic WithLocation(Location location);
```